scipy/linalg/flapack_gen.pyf.src

```diff
@@ -307,7 +307,8 @@ subroutine <prefix2>gesdd(m,n,minmn,u0,u1,vt0,vt1,a,compute_uv,full_matrices,u,s
     integer intent(in),optional,check(full_matrices==0||full_matrices==1):: full_matrices = 1
     integer intent(hide),depend(a):: m = shape(a,0)
     integer intent(hide),depend(a):: n = shape(a,1)
-    integer intent(hide),depend(m,n):: minmn = MIN(m,n)
+    check(m <= sqrt(F_INT_MAX)/n)
+    integer intent(hide),depend(m,n) :: minmn = MIN(m,n)
```

Comment:
Another approach of detecting integer overflow that does not involve introducing `F_INT_MAX` and not using `sqrt`, is (untested):
 ```suggestion
     integer intent(hide),depend(m,n) :: mn_overflow = m * n
     check(m == 0 || mn_overflow / m == n)
     integer intent(hide),depend(m,n) :: minmn = MIN(m,n)
```
or similar.

Notice that when the check fails, the check expression is displayed to the user. So, it is advisable to use an expression that users interpret it correctly.